spark3/src/main/java/org/apache/iceberg/spark/CreateActions.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.connector.catalog.Identifier;
+
+public class CreateActions {
+  private final Identifier newTableName;
+  private final SparkSession spark;
+
+  private CreateActions(SparkSession spark, Identifier newTableName) {
+    this.spark = spark;
+    this.newTableName = newTableName;
+  }
+
+  /**
+   * Create a new Iceberg table in the Iceberg SparkSession catalog based on an
+   * existing non-iceberg table.
+   * @param spark the session to use for interacting with Spark
+   * @param newTableName the string representation of the multipart identifier for new table's name
+   */
+  public static CreateActions createIcebergTable(SparkSession spark, String newTableName) {
+    Identifier identifier = Spark3Util.parseIdentifier(spark, newTableName);
+    if (identifier.namespace() == null || identifier.namespace().length == 0) {
+      identifier = Identifier.of(new String[] {spark.catalog().currentDatabase()}, identifier.name());
+    }
+    return createIcebergTable(spark, identifier);
+  }
+
+  /**
+   * Create a new Iceberg table in the Iceberg SparkSession catalog based on an
+   * existing non-iceberg table.
+   * @param spark the session to use for interacting with Spark
+   * @param newTableName the Spark catalog Identifier for new table's name
+   */
+  public static CreateActions createIcebergTable(SparkSession spark, Identifier newTableName) {
+    return new CreateActions(spark, newTableName);
+  }
+
+  /**
+   * Existing table to be used for creating the Iceberg Table
+   * @param existingTable the string representation of an existing table
+   * @return a Spark Action to perform the migration
+   */
+  public MigrateAction fromSourceTable(String existingTable) {
+    Identifier identifier = Spark3Util.parseIdentifier(spark, existingTable);
+    if (identifier.namespace() == null || identifier.namespace().length == 0) {
+      identifier = Identifier.of(new String[] {spark.catalog().currentDatabase()}, identifier.name());
```

Comment:
I don't think that `spark.catalog().currentDatabase()` is correct. I thought that `spark.catalog()` always returns the built-in v1 catalog.

Reply:
Removed this entirely, we now require a Catalog and Identifier to remove any ambiguity here. We parse this pair from the name as a multipart identifier, see new utility methods.